src/App.tsx

```diff
@@ -1,17 +1,107 @@
-import React from 'react';
-import './App.css';
-
+import React, { useState } from 'react';
+import './App.scss';
+import todos from './api/todos';
 import users from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userSelect, setUserSelect] = useState(0);
+  const [currToddos, setTodos] = useState(preparedTodos);
+  const [click, setClick] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserSelect(Number(event.target.value));
+  };
+
+  const addTodo = () => {
+    const valid = () => (
+      userSelect !== 0 && title !== ''
+    );
+
+    setClick(!click);
+
+    if (valid()) {
+      setTodos(() => {
+        const newTodo = {
+          userId: userSelect,
+          id: currToddos.length + 1,
+          title,
```

Comment:
Yes, I had the same idea, but after thinking twice, I decided to change the way I provide the id, since it will work almost the same way as an index. The task says that you should use the highest id from the already created todos

Reply:
It's mush better way to provide id for this task, but in general it still there's an issue. For example if your app had the reverse button it wouldn't work correct because you take id from the last object in the list, not from the object with the highes id.